src/coreclr/gcinfo/gcinfodumper.cpp

```diff
@@ -288,15 +288,19 @@ PORTABILITY_ASSERT("GcInfoDumper::ReportPointerRecord is not implemented on this
     assert(!"unimplemented on LOONGARCH yet");
     iSPRegister = 0;
 #elif defined(TARGET_RISCV64)
-    assert(!"unimplemented on RISCV64 yet");
-    iSPRegister = 0;
+    iSPRegister = (offsetof(T_CONTEXT, Sp) - offsetof(T_CONTEXT, R0)) / sizeof(ULONGLONG);
 #endif
 
 #if defined(TARGET_ARM) || defined(TARGET_ARM64)
     BYTE* pContext = (BYTE*)&(pRD->volatileCurrContextPointers);
 #elif defined(TARGET_LOONGARCH64)
     assert(!"unimplemented on LOONGARCH yet");
     BYTE* pContext = (BYTE*)pRD->pCurrentContext;
+#elif defined(TARGET_RISCV64)
+    assert(!"unimplemented on RISCV64 yet");
+    // TODO implement risc-v code, that should care about volatile registers (same as arm/arm64 architectures)
```

Comment:
Add `|| defined(TARGET_RISCV64)` to the ARM / ARM64 case above?

Reply:
I am not sure we could change pointer to another size structure here and after this operate with offsets for another structure size in code below.